source/MaterialXGenShader/ShaderStage.cpp

```diff
@@ -70,6 +70,12 @@ ShaderPort* VariableBlock::add(const TypeDesc* type, const string& name, ValuePt
     auto it = _variableMap.find(name);
     if (it != _variableMap.end())
     {
+        // Automatically try to widen the type of the shader port if the requested type differs from
```

Comment:
I assume this also affect / "fix" color if you ask for the same color stream with different number of channels ?